docs/proposals/version_documentation.md

```diff
@@ -0,0 +1,145 @@
+Date: May 2020
+Status: In-review
+---
+
+# Problem
+
+The Thanos website contains the docs area which is built by rendering the markdown files from the tip of master of the Thanos repository. This frequently causes confusion for users, as different Thanos versions have different features.
+
+In this proposal, we want to solve this. In particular, we want to:
+
+1. Build a Version Picker that serves as a tool which will aid easy access to other versions of the documentation. This picker will Include links to change version (the version must be in the URL).
+2. A solid and robust Documentation structure.
+3. Design a workflow for managing docs that integrates with Thanos Git workflow ie updating corresponding docs on pull requests, cherry-picks, etc.
+
+# Motivation
+
+Many Users (mostly developers) would often want to look through the docs of previous releases, but searching for it manually by looking through the entire GitHub repository is time-consuming and ineffective. Because the site is built by rendering the markdown files from the tip of master of the Thanos repository, it causes confusion for users, as breaking changes are often merged into master. The verioning plugin allows the user to access the docs of both latest and previous releases on the Thanos documentation page, and it also allows the developer to fetch, write, and fix the docs of both latest and previous releases.
+
+
+# Requirements
+
+### User Story (Latest)
+
+* As a Thanos developer, I want to be able to just write docs for my current `master` version
+* As a Thanos developer, I want to build website within versioned docs with a single action
+* As a Thanos developer, I don't want to store in a single repository version duplicated docs
+
+### User Story (Previous release)
+
+* As a Thanos developer, I want to be able to fix docs for certain previous release.
+* As a Thanos user, I want to be able to read docs for older version of Thanos
+
+### Use Case
+
+#### Users
+Thanos Developers, Hugo developers, and General Users
+
+#### Precondition
+The user visits the Thanos Documentation Page
+
+#### Basic Course Of Events
+
+1. The User indicates that the site is to show a list of docs (both latest and previous releases) by clicking the version picker (dropdown)
+
+2. The site responds by displaying a list of versions allowing the user to make a selection.
+
+3. User makes selection and the required docs renders on the page/site.
+
+
+#### Postcondition
+
+The site now has a version picker containing a list of versioned docs consisting of both latest and older versions.
+
+
+# Proposed Solution
+
+## 1. Version Picker
+
+Currently, the documentation is residing under the docs/ folder of Thanos repository. It is built by Hugo. It will have a proper drop-down menu just like prometheus [`drop-down menu`](https://prometheus.io/docs/introduction/overview/) which will enable proper versioning. This user facing tool will be built using HTML and CSS
```

Comment:
s/user facing/user-facing/

Reply:
Let's add a final period to the last sentence

Reply:
Small typo, let's change to:
```suggestion
Currently, the documentation resides under the docs/ folder of the Thanos repository. It is built by Hugo. It will have a proper drop-down menu just like prometheus [`drop-down menu`](https://prometheus.io/docs/introduction/overview/) which will enable proper versioning. This use-facing tool will be built using HTML and CSS.
```